modules/fargate/fargate.tf

```diff
@@ -0,0 +1,67 @@
+# Allow Fargate pods and EC2 workers to communicate
+
+resource "aws_security_group_rule" "eks_fargate1" {
+  count                    = local.create ? 1 : 0
+  type                     = "ingress"
+  from_port                = 0
+  to_port                  = 65535
+  protocol                 = "all"
+  security_group_id        = var.worker_security_group_id
+  source_security_group_id = var.cluster_primary_security_group_id
+}
+
+resource "aws_security_group_rule" "eks_fargate2" {
+  count                    = local.create ? 1 : 0
+  type                     = "ingress"
+  from_port                = 0
+  to_port                  = 65535
+  protocol                 = "all"
+  security_group_id        = var.cluster_primary_security_group_id
+  source_security_group_id = var.worker_security_group_id
+}
+
+
+# EKS Fargate Pod Execution Role
+
+data "aws_iam_policy_document" "eks_fargate_pod_assume_role" {
+  count = local.create ? 1 : 0
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["eks-fargate-pods.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "eks_fargate_pod" {
+  count              = local.create ? 1 : 0
+  name               = format("%s-fargate", var.cluster_name)
+  assume_role_policy = join("", data.aws_iam_policy_document.eks_fargate_pod_assume_role.*.json)
+  tags               = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "eks_fargate_pod" {
+  count      = local.create ? 1 : 0
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSFargatePodExecutionRolePolicy"
+  role       = join("", aws_iam_role.eks_fargate_pod.*.name)
+}
+
+
+# EKS Fargate profiles
+
+resource "aws_eks_fargate_profile" "this" {
+  for_each               = var.profiles
+  cluster_name           = var.cluster_name
+  fargate_profile_name   = format("%s-fargate-%s", var.cluster_name, each.value.namespace)
```

Comment:
Great idea, I've added that. But when I tested it, the name was never actually passed down to the module because the object type definition (in `variables.tf`) doesn't specify `name` as a key. I don't want `name` to be a required argument, but an optional argument is [not supported yet](https://github.com/hashicorp/terraform/issues/19898).

A workaround would be to not use the object type definition at all, but then we wouldn't get the automatic type check and the implicit documentation of allowed keys and their types.